### PR TITLE
2.5 metadata fixes

### PR DESCRIPTION
**Pull request by mstanleyjones**
Changes required to pull the reference docs from this repo into the `docker.github.io` repo for use in the live docs. This is against `release/2.5` and includes cherry-picks of two merges which are already in `master, along with one new commit which will also need to be merged into `master` (in a separate PR) when it is finalized.

### CODE REVIEW

**Comment by mstanleyjones** [comment]
PTAL @johndmulhausen @aaronlehmann @stevvooe @thaJeztah

**Comment by johndmulhausen** [comment]
LGTM

**Comment by stevvooe** [comment]
LGTM

Looks like we need to cherry pick a few lint fixups to get the build to work. Let me know if you need help tracking that down.

**Comment by mstanleyjones** [comment]
Hi @stevvooe, these test failures look unrelated to this change. I didn't change any Go files, but all of the errors in the failed test are in Go files. Am I missing something?

**Comment by mstanleyjones** [comment]
OK, with @RichardScothern's help I cherry-picked d35d94dcec76dbed059de99b18626aacc133efd0 underneath my fix. Review is appreciated because I did have to resolve some conflicts.

**Comment by dmcgowan** [comment]
Looks like the conflict resolution was not correct, all the changes related to object ACL in s3 should not be pulled in. Let me know if you want help applying and resolving the cherry pick.

**Comment by mstanleyjones** [comment]
Second try. PTAL @RichardScothern @dmcgowan

File: registry/storage/driver/s3-aws/s3.go [hunk 1]
```diff
@@ -326,13 +326,6 @@ func New(params DriverParameters) (*Driver, error) {
 	// }
 
 	d := &driver{
-		S3:            s3obj,
```
Comment (dmcgowan) [inline registry/storage/driver/s3-aws/s3.go]: This block shouldn't be pulled in

Comment (dmcgowan) [inline registry/storage/driver/s3-aws/s3.go]: Just excluding the s3 changes from the commit should do it

**Comment by mstanleyjones** [comment]
Another try, edited the lint fix to pull out the s3.go file...

**Comment by dmcgowan** [comment]
LGTM

**Pull request merged**